test/debug/list_test.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require_relative '../support/test_case'
+
+module DEBUGGER__
+  class ListTest < TestCase
+    def program
+      <<~RUBY
+       1| class Baz
+       2|   def inspect
+       3|     raise 'Boom'
+       4|   end
+       5| end
+       6|
+       7| baz = Baz.new
+       8| bar = 1
+      RUBY
+    end
+
+    def test_list_lists_the_source_code
+      debug_code(program) do
+        type 'list'
+        assert_line_text(Regexp.new(program))
+
+        type 'q!'
+      end
+    end
+
+    def test_list_only_lists_code_after_given_line
+      debug_code(program) do
+        clear_last_backlog # clear the code displayed at the beginning
```

Comment:
why is it neeed? `@last_backlog` is cleared just after `type`.

Reply:
actually it doesn't (I remember it used to do that though). let me see if doing this will break other tests.